blockstack/blockstackd.py

```diff
@@ -3529,29 +3586,156 @@ def run_blockstackd():
         db = get_db_state(working_dir)
         ch = db.get_consensus_at(block)
         if str(ch) != consensus_hash:
-            print "Fast-sync import does not match consensus hash!"
+            print_status("Fast-sync import does not match consensus hash!")
             sys.exit(1)
 
+        output_dir = os.path.abspath(args.output_path)
+        if not os.path.exists(output_dir):
+            os.makedirs(output_dir)
+
         if args.verify:
             db_verified = False
-            print "Running database verification..."
+            print_status("Running database verification...")
             tmpdir = tempfile.mkdtemp('blockstack-verify-chainstate-XXXXXX')
             try:
                 db_verified = verify_database(consensus_hash, block, working_dir, tmpdir, start_block=FIRST_BLOCK_MAINNET)
             except Exception as e:
-                print "Exception during database verification"
+                print_status("Exception during database verification")
                 print e
                 traceback.print_exc()
             shutil.rmtree(tmpdir)
             if db_verified:
-                print "Database is consistent with %s" % args.consensus_hash
+                print_status("Database is consistent with %s" % args.consensus_hash)
             else:
-                print "Database is NOT CONSISTENT with block {} and consensus hash {}".format(block, args.consensus_hash)
+                print_status("Database is NOT CONSISTENT with block {} and consensus hash {}".format(block, args.consensus_hash))
                 sys.exit(1)
+        
+        print_status("Exporting subdomain metadata to csv...")
+        if not which_tools(['sqlite3']):
+            print_status('Could not find sqlite3 on system path')
+            sys.exit(1)
+        subdomain_db_path = os.path.abspath(os.path.join(working_dir, 'subdomains.db'))
+        subdomain_csv_path = os.path.join(output_dir, 'subdomains.csv')
+        subdomain_csv_hash_path = subdomain_csv_path + '.sha256'
+        subdomain_query_str = 'SELECT zonefile_hash, fully_qualified_subdomain, owner from subdomain_records ORDER BY owner, fully_qualified_subdomain;'
+        cmd = 'sqlite3 {} -cmd ".headers on" -cmd ".mode csv" -cmd ".output {}" "{}"'.format(
+            pipes.quote(subdomain_db_path), 
+            pipes.quote(subdomain_csv_path), 
+            subdomain_query_str)
+        p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        p_res = p.communicate()
+        rc = p.returncode
+        if rc != 0:
+            print_status('Subdomain sqlite data dump failed with error {}: {}'.format(rc, p_res[1]))
+            sys.exit(1)
 
-        print "Querying account addresses..."
+        print_status("Writing subdomains csv file hash...")
+        from hashlib import sha256
+        with open(subdomain_csv_path, 'rb') as f:
+            file_bytes = f.read()
+            file_hash = sha256(file_bytes).hexdigest()
```

Comment:
Did you test this with another hash sum implementation (e.g., https://linux.die.net/man/1/sha256sum)?

Reply:
Yep!
```
$ sha256sum subdomains.csv
67f27ade9d94583ae3324ad37b934911268a22a8c2f625920b50a62b478b2123  subdomains.csv
$ cat subdomains.csv.sha256 
67f27ade9d94583ae3324ad37b934911268a22a8c2f625920b50a62b478b2123
```

```
$ sha256sum chainstate.json
93835ea4acb8ef376e502a2b0092402cb3860619356e88f6019dc0de3a19a759  chainstate.json
$ cat chainstate.json.sha256
93835ea4acb8ef376e502a2b0092402cb3860619356e88f6019dc0de3a19a759
```